app/src/org/commcare/activities/components/ImageCaptureProcessing.java

```diff
@@ -66,17 +67,72 @@ private static Pair<File, String> moveAndScaleImage(File originalImage, boolean
         // Encrypt the scaled or original image to final path
         if (HiddenPreferences.isMediaCaptureEncryptionEnabled()) {
             finalFilePath = finalFilePath + MediaWidget.AES_EXTENSION;
+            File sourceFile = new File(sourcePath);
+            File destFile = new File(finalFilePath);
+            
             try {
+                // Extract EXIF data before encryption
+                ExifInterface sourceExif = null;
+                String mimeType = FileUtil.getMimeType(sourcePath);
+                if (mimeType != null && mimeType.startsWith("image/")) {
+                    sourceExif = new ExifInterface(sourcePath);
+                }
+                
+                // Perform encryption
                 EncryptionIO.encryptFile(sourcePath, finalFilePath, formEntryActivity.getSymetricKey());
+                
+                // If we had EXIF data, store it in a companion file
+                if (sourceExif != null) {
+                    String exifPath = finalFilePath + ".exif";
```

Comment:
I am a bit unsure how it's working. A few questions reg.  encrypted images - 

1. Curious why do we need to create a separate `.exif` file here to retain exif data ?

2. Before uploading images to the server, we decrypt them first by using a [EncryptedFileBody](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/network/EncryptedFileBody.java). Do we also need to upload the `.exif` file there to be able to retain exif data on server ?

3. In your testing, have you been able to test the exif retention for final image on server with encyrption enabled ?